wgpu/src/backend/webgpu.rs

```diff
@@ -37,7 +37,11 @@ unsafe impl<T> Send for Sendable<T> {}
 #[cfg(send_sync)]
 unsafe impl<T> Sync for Sendable<T> {}
 
-pub(crate) struct ContextWebGpu(webgpu_sys::Gpu);
+pub(crate) struct ContextWebGpu {
+    // Invariant: `Some(gpu)` implies that `gpu`` is not `undefined` (i.e., the
+    // browser advertises support for WebGPU).
+    gpu: Option<webgpu_sys::Gpu>,
```

Comment:
wrapping this in an option looks nice, but doesn't change the fact that the value inside the option may _still_ be undefined. Or in other words, the invariant documented here can't be statically enforced, so might as well just assert or handle the undefined cases individually

Reply:
> can't be statically enforced, so might as well just assert

I disagree. The invariant only needs to be enforced in one place (`init()`), and anybody in the future that writes new code trying to use `self.gpu` will be forced by the compiler to deal with the fact that the thing might be undefined.

I can change if you really would rather me do so, but I do believe that writing it this way makes future bugs harder to write.

Reply:
I think that's a fine compromise. Where do you suggest I put that `DefinedJsValue<T>` type?

Also, should it be `DefinedNonNullJsValue`? Or is null-ness not really so much of a problem?

> I'm almost surprised wasm_bindgen doesn't come with it already, but maybe I missed it.

I can't find anything of the sort.

Reply:
I think you're right, we want all the way here - `DefinedNonNullJsValue` is what we need!
I'm a big fan of fine grained modules - makes it easy to move somewhere else when needed. I'd put it in backend/webgpu right now since this is the place where all other webgpu related javascript stuff lives.

Thanks for bearing with me on this!

Reply:
Okay, I added `DefinedNonNullJsValue` and used it in `ContextWebGpu`.

I also used it in the signature of `get_browser_gpu_property()` to try to make it less wonky :)